src/Language/Haskell/Liquid/Bare/Resolve.hs

```diff
@@ -535,11 +537,24 @@ instance ResolveSym Ghc.DataCon where
                     Ghc.AConLike (Ghc.RealDataCon x) -> Just x 
                     _                                -> Nothing
 
-instance ResolveSym F.Symbol where 
-  resolveLocSym env name _ lx = case resolveLocSym env name "Var" lx of 
-    Left _               -> Right (val lx)
+instance ResolveSym F.Symbol where
+  resolveLocSym env name _ lx = case resolveLocSym env name "Var" lx of
+    -- If we can't resolve the input 'Symbol' from an 'Id', try again
+    -- by grabbing the 'Id' of an 'AConLike', if any.
+    -- See test T1688.
+    Left _               -> case resolveLocSym env name "Var" lx of
+      Left _  -> Right (val lx)
+      Right v -> Right (F.symbol $ unWrap (v :: Wrapped Ghc.Var))
```

Comment:
Could we perhaps use this opportunity to actually document what is going on here? Write a comment for the `ResolveSym` class, and describe the entire logic in the `F.Symbol` case?